File: Javascript problems/twoSum.js
/**
 * @param {number[]} nums - An array of integers.
 * @param {number} target - The target integer sum.
 * @return {number[]} - An array containing the indices of the two numbers whose sum is equal to the target.
 */
var twoSum = function(nums, target) {
    // Iterate through the array using 'i' as the index of the first number.
    for (let i = 0; i < nums.length; i++) {
        // Iterate through the array again using 'j' as the index of the second number.
        for (let j = i + 1; j < nums.length; j++) {
            // Check if the sum of the current pair of numbers equals the target.
            if (nums[j] === target - nums[i]) {
                // If a pair is found, return the indices of the two numbers.
                return [i, j];
            }
        }
    }
    // If no pair is found, return an empty array.
    return [];
};

File: Java Problems/twoSum.java
//Given an array of integers nums and an integer target, return indices of the two numbers such that they add up to target.
//You may assume that each input would have exactly one solution, and you may not use the same element twice.
//You can return the answer in any order.

/* Example 1:
Input: nums = [2,7,11,15], target = 9
Output: [0,1]
Explanation: Because nums[0] + nums[1] == 9, we return [0, 1].*/

class solution {
    public int[] twoSum(int[] nums, int target) {
        //loop through each element in the array
        for(int i = 0; i < nums.length; i++) {
            //loopthrough each element staring from the next index of i
            for(int j = 0; i < nums.length; j++) {
                //check if the sum of nums[i] and nums[j] equals the target
                if(nums[j] == target - nums[i]) {
                    return new int[] {i, j};
                }//if
            }//jfor
        }//ifor
    }//twoSum
    //if pair is not found return null
    return null;
}//solution